src/main.cpp

```diff
@@ -200,51 +374,253 @@ uint64_t deepSleepTime = defaultDeepSleepTime; // actual sleep time in minutes,
+    display.setFont(&OpenSansSB_18px);
+    setTextPos("You may see this screen because of lost connection to the WIFi", 10, 40);
+    setTextPos("Please turn off your mobile data connection!", 10, 65);
+    setTextPos("Connect this device to the Internet for first time, complete these steps:", 10, 90);
+
+    centeredText("Connect to the following", DISPLAY_RESOLUTION_X / 4, 170);
+    centeredText("access point", DISPLAY_RESOLUTION_X / 4, 190);
+    centeredText("Open web browser, go to this address", DISPLAY_RESOLUTION_X * 3 / 4, 170);
+    centeredText("and set up connection to your WiFi", DISPLAY_RESOLUTION_X * 3 / 4, 190);
+    centeredText("SSID: " + hostname, DISPLAY_RESOLUTION_X / 4, 400);
+    centeredText("password: zivyobraz", DISPLAY_RESOLUTION_X / 4, 420);
+    centeredText("192.168.4.1", DISPLAY_RESOLUTION_X * 3 / 4, 410);
+    centeredText("In case of any trouble please visit wiki.zivyobraz.eu ", DISPLAY_RESOLUTION_X / 2, DISPLAY_RESOLUTION_Y - 20);
+  } else if (DISPLAY_RESOLUTION_X > 600) {
+    Serial.println("Resolution > 600px,");
+    display.setFont(&OpenSansSB_20px);
+    centeredText("WiFi connection setup", DISPLAY_RESOLUTION_X / 2, 15);
+    centeredText("Step 1:", DISPLAY_RESOLUTION_X / 4, 110);
+    centeredText("Step 2:", DISPLAY_RESOLUTION_X * 3 / 4, 110);
+    display.setFont(&OpenSansSB_14px);
+    setTextPos("You may see this screen because of lost connection to the WIFi", 10, 40);
+    setTextPos("Please turn off your mobile data connection!", 10, 60);
+    setTextPos("Connect this device to the Internet for first time, complete these steps:", 10, 80);
+
+    centeredText("Connect to the following", DISPLAY_RESOLUTION_X / 4, 140);
+    centeredText("access point", DISPLAY_RESOLUTION_X / 4, 160);
+    centeredText("Open web browser, go to this address", DISPLAY_RESOLUTION_X * 3 / 4, 140);
+    centeredText("and set up connection to your WiFi", DISPLAY_RESOLUTION_X * 3 / 4, 160);
+    centeredText("SSID: " + hostname, DISPLAY_RESOLUTION_X / 4, 300);
+    centeredText("password: zivyobraz", DISPLAY_RESOLUTION_X / 4, 320);
+    centeredText("192.168.4.1", DISPLAY_RESOLUTION_X * 3 / 4, 310);
+    centeredText("In case of any trouble please visit wiki.zivyobraz.eu ", DISPLAY_RESOLUTION_X / 2, DISPLAY_RESOLUTION_Y - 20);
+  } else if (DISPLAY_RESOLUTION_X >= 400) {
+    //TODO for 4" display
+  } else if (DISPLAY_RESOLUTION_X < 400) {
+    //TODO for 2.13" display
+  } else {
+    //some special case
+  }
+
+  // Create the QR code
+
+  /*QR code hint
+
+    Common format: WIFI:S:<SSID>;T:<WEP|WPA|nopass>;P:<PASSWORD>;H:<true|false|blank>;;
+
+    Sample: WIFI:S:MySSID;T:WPA;P:MyPassW0rd;;
+
+  */
+
+  String qrString = "WIFI:S:";
+  qrString += hostname;
+  qrString += ";T:WPA;P:zivyobraz;;";
+  //Serial.println(qrString);
+
+  if (DISPLAY_RESOLUTION_X >= 800) {
+    drawQrCode(qrString.c_str(), 4, (DISPLAY_RESOLUTION_Y / 2) + 50, DISPLAY_RESOLUTION_X / 4, 4);
```

Comment:
Same as above

Reply:
I already found some things you are pointing out. This is chaos for me - I need to deal with conflicts first, merge this PR and then maybe improve code. Right now, I’m dealing with too many things at once.